src/console_internal.h

```diff
@@ -53,25 +54,31 @@ struct IConsoleCmd {
  * - ";" allows for combining commands (see example 'ng')
  */
 struct IConsoleAlias {
-	char *name;                 ///< name of the alias
-	IConsoleAlias *next;        ///< next alias in list
+	IConsoleAlias(std::string name, std::string cmdline) : name(name), cmdline(cmdline) {}
 
-	char *cmdline;              ///< command(s) that is/are being aliased
+	std::string name;           ///< name of the alias
+	std::string cmdline;        ///< command(s) that is/are being aliased
 };
 
-/* console parser */
-extern IConsoleCmd   *_iconsole_cmds;    ///< List of registered commands.
-extern IConsoleAlias *_iconsole_aliases; ///< List of registered aliases.
+struct IConsole
+{
+	typedef std::map<std::string, IConsoleCmd> CommandList;
+	typedef std::map<std::string, IConsoleAlias> AliasList;
+
+	/* console parser */
+	static CommandList &Commands();
+	static AliasList &Aliases();
+
+	/* Commands */
+	static void CmdRegister(std::string name, IConsoleCmdProc *proc, IConsoleHook *hook = nullptr);
```

Comment:
This register and the AliasRegister functions could use const std::string &name. For the get functions that is not beneficial as they need a copy internally to remove the underscores from the string.